packages/ui/src/components/Toggle/index.tsx

```diff
@@ -202,11 +226,20 @@ export const Toggle = forwardRef(
               ref={ref}
             />
           </StyledToggle>
-          {label && labelPosition === 'right' ? (
-            <>
-              {label}
-              {required ? <RequiredIcon /> : null}
-            </>
+          {labelPosition === 'right' ? (
+            <Stack gap={0.25} alignItems="baseline">
+              {label ? (
+                <Row templateColumns="auto 1fr" gap={1} alignItems="center">
+                  {label}
+                  {required ? <RequiredIcon /> : null}
+                </Row>
+              ) : null}
+              {helper ? (
+                <Text as="p" variant="bodySmall" prominence="weak">
+                  {helper}
+                </Text>
+              ) : null}
+            </Stack>
```

Comment:
Mb we should add an optional children prop to allow proper alignment of a content to the label. It's especially useful when content depends of the toggle value